go/vt/mysqlctl/xtrabackupengine.go

```diff
@@ -0,0 +1,484 @@
+// the appropriate parameters
+type XtrabackupEngine struct {
+}
+
+var (
+	// path where backup engine program is located
+	xtrabackupEnginePath = flag.String("xtrabackup_root_path", "", "directory location of the xtrabackup executable, e.g., /usr/bin")
+	// flags to pass through to backup engine
+	xtrabackupBackupFlags = flag.String("xtrabackup_backup_flags", "", "flags to pass to backup command. these will be added to the end of the command")
+	// flags to pass through to restore phase
+	xbstreamRestoreFlags = flag.String("xbstream_restore_flags", "", "flags to pass to xbstream command during restore. these will be added to the end of the command. These need to match the ones used for backup e.g. --compress / --decompress, --encrypt / --decrypt")
+	// streaming mode
+	xtrabackupStreamMode = flag.String("xtrabackup_stream_mode", "tar", "which mode to use if streaming, valid values are tar and xbstream")
+	xtrabackupUser       = flag.String("xtrabackup_user", "", "User that xtrabackup will use to connect to the database server. This user must have all necessary privileges. For details, please refer to xtrabackup documentation.")
+)
+
+const (
+	streamModeTar = "tar"
+	xtrabackup    = "xtrabackup"
+	xbstream      = "xbstream"
+	successMsg    = "completed OK!"
+)
+
+// XtraBackupManifest represents the backup.
+type XtraBackupManifest struct {
+	// FileName is the name of the backup file
+	FileName string
+	// BackupMethod, set to xtrabackup
+	BackupMethod string
+	// Position at which the backup was taken
+	Position mysql.Position
+	// SkipCompress can be set if the backup files were not run
+	// through gzip.
+	SkipCompress bool
+	// Params are the parameters that backup was run with
+	Params string `json:"ExtraCommandLineParams"`
+}
+
+func (be *XtrabackupEngine) backupFileName() string {
+	fileName := "backup"
+	if *xtrabackupStreamMode != "" {
+		fileName += "."
+		fileName += *xtrabackupStreamMode
+	}
+	if *backupStorageCompress {
+		fileName += ".gz"
+	}
+	return fileName
+}
+
+// ExecuteBackup returns a boolean that indicates if the backup is usable,
+// and an overall error.
+func (be *XtrabackupEngine) ExecuteBackup(ctx context.Context, cnf *Mycnf, mysqld MysqlDaemon, logger logutil.Logger, bh backupstorage.BackupHandle, backupConcurrency int, hookExtraEnv map[string]string) (bool, error) {
+
+	if *xtrabackupUser == "" {
+		return false, vterrors.New(vtrpc.Code_INVALID_ARGUMENT, "xtrabackupUser must be specified.")
+	}
+	conn, err := mysqld.GetDbaConnection()
+	defer conn.Close()
+	if err != nil {
+		return false, vterrors.Wrap(err, "unable to obtain a connection to the database")
+	}
+	pos, err := conn.MasterPosition()
+	if err != nil {
+		return false, vterrors.Wrap(err, "unable to obtain master position")
+	}
+	flavor := pos.GTIDSet.Flavor()
+	logger.Infof("Detected MySQL flavor: %v", flavor)
+
+	backupProgram := path.Join(*xtrabackupEnginePath, xtrabackup)
+
+	flagsToExec := []string{"--defaults-file=" + cnf.path,
```

Comment:
why not just split the args and not use =?

```go
[]string{"--defaults-file", cnf.path,
"--backup",
"--socket", cnf.SocketFile
...
```

Reply:
This is how xtrabackup expects us to provide the args.